### PR TITLE
PR Version 2.2.0.0

### PR DESCRIPTION
**Pull request by snapp1**

### CODE REVIEW

File: test/visualTest.ts [hunk 1]
```diff
@@ -50,675 +46,694 @@ import areColorsEqual = helpers.areColorsEqual;
...
-            ): boolean {
-                return elements.some((element: JQuery) => {
-                    const currentColor: string = element.css(colorStyleName);
+    beforeEach(() => {
+      hightlightedElementNumber = Math.round(getRandomNumber(0, elementCount - 1));
 
-                    if (!currentColor || !color) {
-                        return currentColor === color;
-                    }
+      dataViewWithHighLighted = defaultDataViewBuilder.getDataView(undefined, true, hightlightedElementNumber);
```
Comment (MulyukovAidar) [inline test/visualTest.ts:725]: try calling visualBuilder.update() to eliminate timeout awaiting

**Review by MulyukovAidar — CHANGES_REQUESTED** [review]

File: src/settings.ts [hunk 2]
```diff
@@ -24,54 +24,246 @@
...
+            minValue: {
+                type: powerbi.visuals.ValidatorType.Min,
+                value: 1,
+            },
+            maxValue: {
+                type: powerbi.visuals.ValidatorType.Max,
+                value: 100,
+            },
+            unitSymbolAfterInput: true,
+            unitSymbol: "%"
```
[the remark this answers is not in the document]
Comment (snapp1) [reply src/settings.ts:288]: No

File: src/WordCloud.ts [hunk 3]
```diff
@@ -880,6 +878,7 @@ export class WordCloud implements IVisual {
         }
 
         this.eventService.renderingStarted(visualUpdateOptions);
+        this.formattingSettings = this.formattingSettingsService.populateFormattingSettingsModel(WordCloudSettings, visualUpdateOptions.dataViews);
```
[the remark this answers is not in the document]
Comment (snapp1) [reply src/WordCloud.ts:877]: removed code as it was always giving true

File: src/WordCloud.ts [hunk 4]
```diff
@@ -784,48 +778,53 @@ export class WordCloud implements IVisual {
...
-        this.root.on('contextmenu', () => {
-            const mouseEvent: MouseEvent = getEvent();
-            const eventTarget: EventTarget = mouseEvent.target;
-            let dataPoint: any = d3.select(<d3.BaseType>eventTarget).datum();
-            this.selectionManager.showContextMenu(dataPoint ? dataPoint.selectionId : {}, {
-                x: mouseEvent.clientX,
-                y: mouseEvent.clientY
-            });
-            mouseEvent.preventDefault();
+        this.root.on('contextmenu', (event) => {
```
[the remark this answers is not in the document]
Comment (snapp1) [reply src/WordCloud.ts]: done

File: src/WordCloud.ts [hunk 5]
```diff
@@ -493,44 +490,42 @@ export class WordCloud implements IVisual {
         };
     }
 
-    public static PARSE_SETTINGS(dataView: DataView, colorHelper: ColorHelper): WordCloudSettings {
-        const settings: WordCloudSettings = WordCloudSettings.parse<WordCloudSettings>(dataView);
-
-        settings.general.minFontSize = Math.max(
-            settings.general.minFontSize,
+    public static PARSE_SETTINGS(settings: WordCloudSettings, colorHelper: ColorHelper): WordCloudSettings {
```
[the remark this answers is not in the document]
Comment (snapp1) [reply src/WordCloud.ts]: done

File: src/WordCloud.ts [hunk 6]
```diff
@@ -403,32 +400,32 @@ export class WordCloud implements IVisual {
     private static punctuationRegExp: RegExp = new RegExp(`[${WordCloud.Punctuation.join("\\")}]`, "gim");
     private static whiteSpaceRegExp: RegExp = /\s/;
 
-
     public static CONVERTER(
```
[the remark this answers is not in the document]
Comment (snapp1) [reply src/WordCloud.ts]: done

File: capabilities.json [hunk 7]
```diff
@@ -95,7 +96,7 @@
                     }
                 },
                 "minFontSize": {
-                    "displayName": "Min font size",
```
[the remark this answers is not in the document]
Comment (snapp1) [reply capabilities.json:98]: done

File: package.json [hunk 8]
```diff
@@ -28,54 +29,75 @@
...
+    "@types/lodash.toarray": "^4.4.7",
+    "@types/lodash.uniqby": "^4.7.7",
+    "@types/webpack": "^5.28.1",
+    "@typescript-eslint/eslint-plugin": "^6.1.0",
+    "@typescript-eslint/parser": "^6.1.0",
+    "css-loader": "^6.8.1",
+    "eslint": "^8.45.0",
+    "eslint-plugin-powerbi-visuals": "^0.8.1",
+    "jasmine": "^5.0.2",
+    "karma": "^6.4.2",
```
[the remark this answers is not in the document]
Comment (snapp1) [reply package.json:82]: some of them is requred, some is not. Which are not required - removed

File: src/settings.ts [hunk 9]
```diff
@@ -24,54 +24,246 @@
...
+                value: 1,
+            },
+            maxValue: {
+                type: powerbi.visuals.ValidatorType.Max,
+                value: 100,
+            },
+        },
+    });
+
+    public isBrokenText = new formattingSettings.ToggleSwitch({
```
[the remark this answers is not in the document]
Comment (snapp1) [reply src/settings.ts]: done

File: test/visualTest.ts (hunk 1 above)
Comment (snapp1) [reply test/visualTest.ts:725]: no luck

**Pull request merged**